private/engine/config.go

```diff
@@ -1,52 +1,121 @@
 package engine
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/BurntSushi/toml"
 )
 
+type connectionType uint
+
+const (
+	socketConnection connectionType = iota
+	httpConnection
+)
+
 type Config struct {
-	Socket  string `toml:"socket"`  // socket filename
-	WorkDir string `toml:"workdir"` // path to socket file
+	ConnectionType connectionType
+	SocketConfig   socketConfig `toml:",omitempty"`
+	HttpConfig     httpConfig   `toml:",omitempty"`
+}
 
-	DialTimeout           uint // timeout for connecting to socket (seconds)
-	RequestTimeout        uint // timeout for writing to socket (seconds)
-	ResponseHeaderTimeout uint // timeout for reading from socket (seconds)
+type socketConfig struct {
+	Socket                string // socket filename
+	WorkDir               string // path to socket file
+	DialTimeout           uint   // timeout for connecting to socket (seconds)
+	RequestTimeout        uint   // timeout for writing to socket (seconds)
+	ResponseHeaderTimeout uint   // timeout for reading from socket (seconds)
+}
 
-	// Deprecated
-	SocketPath string `toml:"socketPath"`
+type httpConfig struct {
+	Url             string // transaction manager URL for HTTP connection
+	Tls             string // if "OFF" then TLS disabled, if "STRICT" then TLS enabled
```

Comment:
Modes `off` and `verify-full` are implemented but called `none` and `strict` to match the naming using in the Tessera config (there isn't any standard naming convention, but none/strict is used in other places e.g. by mysql).

Rather than add a `require` mode, I've added `insecureSkipVerify` as a separate config option, which is consistent with the security plugin and other TLS-related work we’ve done for Quorum.

I didn't add `verify-ca`, as I'm not sure it adds a lot of benefit and as per your statement, it would not be straightforward to implement.